src/base/RegexPattern.h

```diff
@@ -22,26 +23,42 @@ class RegexPattern
 
 public:
     RegexPattern() = delete;
-    RegexPattern(int aFlags, const char *aPattern);
+    RegexPattern(const SBuf &aPattern, int aFlags);
     ~RegexPattern();
 
-    // regex type varies by library, usually not safe to copy
-    RegexPattern(const RegexPattern &) = delete;
-    RegexPattern &operator =(const RegexPattern &) = delete;
+    RegexPattern(RegexPattern &&) = delete; // no copying of any kind
 
-    RegexPattern(RegexPattern &&);
-    RegexPattern &operator =(RegexPattern &&);
+    /// whether the regex differentiates letter case
+    bool caseSensitive() const { return !(flags & REG_ICASE); }
+
+    /// Whether this is an "any single character" regex ("."). In some contexts,
+    /// that regex is (ab)used as a special "should match anything" default.
+    bool isDot() const { return pattern.length() == 1 && pattern[0] == '.'; }
```

Comment:
FYI; In GNU regex there are implicit prefix/suffix wildcards in the absence of an anchor. The pattern (.) actually means "any 1-or-more characters". To match a specific character count requires anchors. i.e. the pattern (^.$) is what means "any single character"

Are there specific cases of "abuse" you can point at?

Reply:
We seem to be in agreement on the technical details. That clears up my worry about the technical side, thank you. The below I do not insist on:

My issue is documenting it as "abuse" to use it as a default. IMO the existing DefaultRule is the broken part - it matches non-existent URLs and ACL values.

IMO that second sentence should be dropped and (if anything) a TODO added to DefaultRule.

Reply:
I dropped the second sentence in afba89c00083223961305981e7fa2d0f28a5886e

I am pretty sure it is abuse (as I tried to prove but apparently failed), but the sentence is not critical. My primary motivation for documenting this abuse was to reduce the temptation to reuse this method or this approach in new code.

I agree that the DefaultRule is also problematic. However, it is not really being abused IMO -- we just lack a feature to make those default refresh parameters configurable explicitly (i.e. without using a `.` regex that Squid then need to find and treat specially).